dandischema/digests/zarr.py

```diff
@@ -109,7 +109,7 @@ def aggregate_digest(self, checksums: ZarrChecksums) -> str:
         """Generate an aggregated digest for a list of ZarrChecksums."""
         # Use the most compact separators possible
         # content = json.dumps([asdict(zarr_md5) for zarr_md5 in checksums], separators=(',', ':'))0
-        content = checksums.json(**ENCODING_KWARGS)  # type: ignore[arg-type]
+        content = checksums.model_dump_json()
```

Comment:
The `ENCODING_KWARGS` are actually important here, as using the correct separators is necessary to get the right MD5 digest.  See https://github.com/dandi/zarr_checksum/pull/23 for how this was addressed in `zarr_checksum`.

Reply:
In Pydantic V2, `BasModel.model_dump_json` uses the most compact separators by default, and it doesn't pass any extra keyword arguments to `json.dumps()`.

If you would like to be absolutely certain, without relying on the default behavior of `BasModel.model_dump_json`. We can do the following instead.


```py
content = json.dumps(checksums.model_dump(mode='json'), **ENCODING_KWARGS)
```

Reply:
> In Pydantic V2, `BasModel.model_dump_json` uses the most compact separators by default

Where is that documented?  If it's not documented, we can't rely on it.

Reply:
I tested it, but it is not documented. I will use the following then

```py
content = json.dumps(checksums.model_dump(mode='json'), **ENCODING_KWARGS)
```